fvm/src/syscalls/bind.rs

```diff
@@ -144,20 +169,25 @@ macro_rules! impl_bind_syscalls {
                         }
 
                         let ctx = Context{kernel: &mut data.kernel, memory: &mut memory};
-                        Ok(match syscall(ctx $(, $t)*).into()? {
+                        let result = match syscall(ctx $(, $t)*).into()? {
                             Ok(value) => {
                                 log::trace!("syscall {}::{}: ok", module, name);
                                 unsafe { *(memory.as_mut_ptr().offset(ret as isize) as *mut Ret::Value) = value };
                                 data.last_error = None;
+                                charge_exec_units_for_gas(&mut caller)?;
                                 0
                             },
                             Err(err) => {
                                 let code = err.1;
                                 log::trace!("syscall {}::{}: fail ({})", module, name, code as u32);
                                 data.last_error = Some(backtrace::Cause::new(module, name, err));
+                                charge_exec_units_for_gas(&mut caller)?;
                                 code as u32
                             },
-                        })
+                        };
+
+                        charge_exec_units_for_gas(&mut caller)?;
```

Comment:
we're calling this twice.

Reply:
oops don't write code while falling asleep in a car